src/NRediSearch/Client.cs

```diff
@@ -562,43 +675,109 @@ public async Task<bool> DeleteSuggestionAsync(string value)
         /// Get completion suggestions for a prefix
         /// </summary>
         /// <param name="prefix">the prefix to complete on</param>
-        /// <param name="fuzzy"> if set,we do a fuzzy prefix search, including prefixes at levenshtein distance of 1 from the prefix sent</param>
-        /// <param name="max">If set, we limit the results to a maximum of num. (Note: The default is 5, and the number cannot be greater than 10).</param>
+        /// <param name="suggestionOptions"> the options on what you need returned and other usage</param>
         /// <returns>a list of the top suggestions matching the prefix</returns>
-        public string[] GetSuggestions(string prefix, bool fuzzy = false, int max = 5)
+        public Suggestion[] GetSuggestions(string prefix, SuggestionOptions options)
```

Comment:
no problem with having a structured API, but the old API should be retained and *proxy* to the new one, if that is at all achievable

Reply:
Done!